engine/src/flutter/lib/ui/painting/display_list_deferred_image_gpu_impeller.cc

```diff
@@ -119,40 +121,27 @@ DlDeferredImageGPUImpeller::ImageWrapper::ImageWrapper(
     fml::TaskRunnerAffineWeakPtr<SnapshotDelegate> snapshot_delegate,
     fml::RefPtr<fml::TaskRunner> raster_task_runner)
     : size_(size),
-      display_list_(std::move(display_list)),
       snapshot_delegate_(std::move(snapshot_delegate)),
       raster_task_runner_(std::move(raster_task_runner)) {}
 
-DlDeferredImageGPUImpeller::ImageWrapper::~ImageWrapper() {
-  fml::TaskRunner::RunNowOrPostTask(
-      raster_task_runner_, [id = reinterpret_cast<uintptr_t>(this),
-                            texture_registry = std::move(texture_registry_)]() {
-        if (texture_registry) {
-          texture_registry->UnregisterContextListener(id);
-        }
-      });
-}
+DlDeferredImageGPUImpeller::ImageWrapper::~ImageWrapper() = default;
 
-void DlDeferredImageGPUImpeller::ImageWrapper::OnGrContextCreated() {
-  FML_DCHECK(raster_task_runner_->RunsTasksOnCurrentThread());
-  SnapshotDisplayList();
-}
+void DlDeferredImageGPUImpeller::ImageWrapper::OnGrContextCreated() {}
 
-void DlDeferredImageGPUImpeller::ImageWrapper::OnGrContextDestroyed() {
-  // Impeller textures do not have threading requirements for deletion, and
-  texture_.reset();
-}
+void DlDeferredImageGPUImpeller::ImageWrapper::OnGrContextDestroyed() {}
 
 bool DlDeferredImageGPUImpeller::ImageWrapper::isTextureBacked() const {
   return texture_ && texture_->IsValid();
 }
 
 void DlDeferredImageGPUImpeller::ImageWrapper::SnapshotDisplayList(
+    sk_sp<DisplayList> display_list,
```

Comment:
Pass the argument as a `std::variant` to indicate that this takes either a `DisplayList` or a `LayerTree` but not both.